presto-main/src/main/java/com/facebook/presto/util/HashCollisionsEstimator.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.util;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.Math.ceil;
+import static java.lang.Math.floor;
+
+public final class HashCollisionsEstimator
+{
+    private static final int NUMBER_OF_ESTIMATES = 500;
+    private static final int NUMBER_OF_VALUES = 10000;
+    private static final double[] COLLISION_ESTIMATES = new double[] {
```

Comment:
You can derive closed formula only for hash tables where next hash table positions are randomized in case of collision. I don't think that there exists formula for hash tables where next candidate position is not random (e.g. +1). I tried to derive one and looked on the Internet, but couldn't find any.